modules/internal/ChapelArray.chpl

```diff
@@ -1767,6 +1767,7 @@ module ChapelArray {
                                                          isAssociativeDom(d2)) {
     if d1._value == d2._value then return true;
     if d1.numIndices != d2.numIndices then return false;
+    // TODO -- should this be a forall?
```

Comment:
You can't return within a forall, so there would need to be some sort of reduction, atomic, or eureka here to make that work...